File: getsurei.pl
# -*- coding: utf-8 -*-
# Author: nakinor
# Created: 2011-12-17
# Revised: 2016-01-29

# 月齢を計算して求めるプログラム

@a_list = (0, 2, 0, 2, 2, 4, 5, 6, 7, 8, 9, 10);
($day, $month, $year) = (localtime(time))[3..5];
$year += 1900;
my $a = (($year - 11) % 19) * 11;
my $b = ($a_list[$month]);
my $c = $day;
$month += 1;     # b は求めてしまったので破壊代入をしちゃう
my $getsurei = ($a + $b + $c) % 30;

print "今日は$year年$month月$day日です。";
print "月齢は約$getsurei日です。\n";

File: getsurei.exs
# Author: nakinor
# Created: 2018-03-02
# Revised: 2018-03-02

# 月齢を計算して求めるプログラム

today = elem(:calendar.local_time(), 0)
year = elem(today, 0)
month = elem(today, 1)
day = elem(today, 2)
a_map = %{1=>0, 2=>2, 3=>0, 4=>2, 5=>2, 6=>4,
          7=>5, 8=>6, 9=>7, 10=>8, 11=>9, 12=>10}

a = rem((year - 11),  19) * 11
b = a_map[month]
c = day
getsurei = rem((a + b + c), 30)

IO.write "今日は#{year}年#{month}月#{day}日です。"
IO.puts "月齢は約#{getsurei}日です。"
